azure-functions-powershell-library/WorkerIndexingHelper.cs

```diff
@@ -0,0 +1,197 @@
+
+            return thisFunction;
+        }
+        private static FunctionInformation CreateRpcMetadataFromFunctionAst(string powerShellFile, FunctionDefinitionAst potentialFunction)
+        {
+            FunctionInformation thisFunction = new FunctionInformation();
+
+            thisFunction.Directory = new FileInfo(powerShellFile).Directory!.FullName;
+            thisFunction.ScriptFile = powerShellFile;
+            thisFunction.Name = potentialFunction.Name;
+            thisFunction.EntryPoint = potentialFunction.Name;
+
+            thisFunction.FunctionId = Guid.NewGuid().ToString();
+
+            ParamBlockAst paramBlock = (ParamBlockAst)potentialFunction.Find(x => x is ParamBlockAst, true);
+            ExtractBindings(thisFunction, paramBlock);
+
+            return thisFunction;
+        }
+
+        private static List<BindingInformation> GetInputBindingInfo(ParamBlockAst paramBlock)
+        {
+            List<BindingInformation> outputBindingInfo = new List<BindingInformation>();
+            foreach (ParameterAst parameter in paramBlock.Parameters)
+            {
+                foreach (AttributeAst attribute in parameter.Attributes)
+                {
+                    BindingInformation? bindingInfo = BindingExtractor.ExtractInputBinding(attribute, parameter);
+                    if (bindingInfo is not null)
+                    {
+                        outputBindingInfo.Add(bindingInfo);
+                    }
+                }
+            }
+            return outputBindingInfo;
+        }
+
+        private static List<BindingInformation> GetOutputBindingInfo(ReadOnlyCollection<AttributeAst> attributes)
+        {
+            List<BindingInformation> outputBindingInfo = new List<BindingInformation>();
+            foreach (AttributeAst attribute in attributes)
+            {
+                BindingInformation? bindingInformation = BindingExtractor.ExtractOutputBinding(attribute);
+                if (bindingInformation is not null)
+                {
+                    outputBindingInfo.Add(bindingInformation);
+                }
+            }
+            return outputBindingInfo;
+        }
+
+        private static void ExtractBindings(FunctionInformation thisFunction, ParamBlockAst paramBlock)
+        {
+            if (paramBlock == null)
+            {
+                return;
+            }
+
+            var functionAttribute = paramBlock.Attributes.Where(x => x.TypeName.Name == "Function" && x.PositionalArguments.Count > 0);
+            if (functionAttribute.Any() && functionAttribute.First().PositionalArguments[0].GetType() == typeof(StringConstantExpressionAst))
+            {
+                thisFunction.Name = ((StringConstantExpressionAst)functionAttribute.First().PositionalArguments[0]).Value;
+            }
+
+            //Input bindings first
+            //thisFunction.Bindings.AddRange(GetInputBindingInfo(paramBlock));
```

Comment:
can this comment be removed?

Reply:
Already done - It was a carryover from when this code was in the worker and there was way more logic here